hello-pepr-watch/capabilities/watch.e2e.test.ts

```diff
@@ -0,0 +1,126 @@
+    data: {
+      name: `${name}-updated`,
+    },
+  });
+}
+
+// deleteCM deletes a ConfigMap by a given name and namespace
+const deleteCM = async (name: string, namespace: string) => {
+  await K8s(kind.ConfigMap).InNamespace(namespace).Delete(name)
+}
+
+// untilUpdatedCM checks if a ConfigMap has been updated by checking the data field
+const untilUpdatedCM = async (name: string, namespace: string): Promise<boolean> => {
+  try {
+    const cm = await K8s(kind.ConfigMap).InNamespace(namespace).Get(name)
+   
+    if (cm.data?.name === `${name}-updated`) {
+      return Promise.resolve(true)
+    } else {
+      return Promise.resolve(false)
+    }
+  }
+  catch (e) {
+    return Promise.resolve(false)
+  }
+}
+describe("watch.ts", () => {
+  beforeAll(async () => await moduleUp(), mins(2));
+  afterAll(async () => await moduleDown(), mins(2))
+
+  describe("tests watch module event phases", () => {
+    let logz: string[] = [];
+    beforeAll(async () => {
+      const file = `${trc.root()}/capabilities/watch.config.yaml`;
+      await timed(`load: ${file}`, async () => {
+        const resources = await trc.load(file);
+        await fullCreate(resources, kind);
+        await untilLogged("Observed creation of one");
+        logz = await logs();
+      });
+    }, mins(1));
+
+    it("should execute callback based on event create phase", () => {
+      const results = logz.filter(l => l.includes("Observed creation of"))
+      expect(results[0]).toContain("one")
+      expect(results[1]).toContain("two")
+      expect(results[2]).toContain("three")
+      expect(results[3]).toContain("four")
+      expect(results[4]).toContain("five")
+    })
+
+    it("should execute callback based on event create-or-update phase", async () => {
+      // Create configMap with name "new"
+      await createUpdateConfigMap("new")
+      // Update configMap with name "two"
+      await createUpdateConfigMap("two")
+      await untilTrue(async () => await untilUpdatedCM("two", "pepr-demo"))
+      await untilLogged("Observed MODIFIED of");
+      logz = await logs();
+
+      // get logs for createdOrUpdated
+      const results = logz.filter(l => l.includes("Observed ADDED of new") || l.includes("Observed MODIFIED of two"))
+
+      let updatedTwo: boolean = false;
+      let createdNew: boolean = false;
+      for (const result of results) {
+        if (result.includes("ADDED of new")) {
+          createdNew = true
+        }
+        if (result.includes("MODIFIED of two")) {
+          updatedTwo = true
+        }
+      }
+
+      // Ensure both created and updated events are logged
+      expect(createdNew).toBe(true)
+      expect(updatedTwo).toBe(true)
+    })
+
+    it("should execute callback based on update phase", async () => {
+      let updated: boolean = false
+      await createUpdateConfigMap("three")
+      updated = await untilUpdatedCM("three", "pepr-demo")
+      await untilTrue(async () => await untilUpdatedCM("three", "pepr-demo"))
```

Comment:
```suggestion
      await untilTrue(() => untilUpdatedCM("three", "pepr-demo"))
```

Reply:
check

Reply:
Is it just my GH UI that makes it look like the suggested change didn't take?  Looks like the inner await statement is still in there..?